### PR TITLE
New User Options: DenySetIdent, DenySetNetwork, DenySetRealName, DenySetQuitMsg, DenySetCTCPReplies

### LINKED ISSUE #1682: [Feature Request] Add ability do disallow normal users to change their ident, add/remove/edit networks/servers and quitmsg
After a brief discussion with MetaNova in IRC and advised by him, i decided to open this feature request.

This feature was implemented by PanicBNC [here](https://github.com/PanicBNC-Dev/ZNC) with a special flag named `Premium`.

##### Small IRC excerpt:
```
[00:10:22] <EliteGod> I'm heading to bed. I'm just leaving a small request here: I've used a PanicBNC custom build for some time. in that build normal users couldn't change their ident, add/remove networks/servers (via IRC or webpanel). is there anyone interested in adapting those changes to znc 1.7.4 in exchange of some sort of Patreon or something like that?
[00:11:26] <MetaNova> if you open a ticket (feature request) I'll put money towards a bounty on it
[00:12:31] <EliteGod> I'll do it tomorrow for sure. do you need the PanicBNC build (they have a GitHub one) to see the changes they've made?
[00:13:04] <EliteGod> it would be just awesome for people that want to provide a public ZNC service
[00:13:08] <MetaNova> relevant links to examples of the implementation elsewhere should be included in the ticket
[00:13:20] <MetaNova> for reference
[00:14:25] <MetaNova> and, yes, I agree: having the ability to prevent users from changing various aspects about their user (ident response, realname, quitmsg, and even the ability to add/remove networks/servers) be part of native ZNC would simplify a good many providers' lives
```

As stated by MetaNova this would help providers and simplify their lives, as probably most of users that run ZNC personally have only one user with all the networks they need.

##### What i think that should be blocked by default:
Deny SetBindHost (checked by default)
Ability to change the ident
Ability to add/remove/edit servers/networks
Ability to change realname
Ability to change quitmsg

All of this should be blocked via IRC commands/webpanel.

@magnum7385 
Feel free to put the bounty on this and add your ideas.

PS: Sorry if my English is bad but i'm Portuguese

### PR DESCRIPTION
**Pull request by bashgeek**
Closes #1682 

Adds the following User settings:

* `DenySetIdent` - Denies setting ident
* `DenySetNetwork` - Denies adding/removing networks/servers
* `DenySetRealName` - Denies setting realname
* `DenySetQuitMsg` - Denies setting quitmsg
* `DenySetCTCPReplies` - Denies adding/removing CTCP replies

The legwork was done in #1796 already, I made some, mostly visual, optimisations in the webadmin module for this:

* Instead of readonly, use disabled text fields
* Properly disable CTCP Reply fields (JS & non-JS version)
* Properly disable Server fields (JS & non-JS version)
* Show a proper message if a user with `DenySetNetwork` has no networks added yet instead of showing an empty table
* Fixed delete network button not working/showing for admins

I also offer to update the wiki with the new settings accordingly, once this gets accepted and merged.

### CODE REVIEW

**Comment by PeGaSuS-Coder** [comment]
Small question: do we have the same counterpart commands via IRC?
I mean, can we use something like `/msg *controlpanel [args]`?

Cheers

**Comment by bashgeek** [comment]
> Small question: do we have the same counterpart commands via IRC?
> I mean, can we use something like /msg *controlpanel [args]?

Yes, this PR includes webadmin and controlpanel.

> Also, I suppose admins do have override rights without the need to fiddle with anything?
Of course, admins can always update anything.

File: modules/data/webadmin/tmpl/add_edit_user.tmpl [hunk 1]
```diff
@@ -129,43 +129,47 @@
...
-				<table class="sortable">
+				<? IF !NetworkEdit && !NetworkLoop ?>
+					<? FORMAT "No networks have been added for you yet." ?>
+				<? ELSE ?>
+				<table class="sortable networks" data-network-edit="<? IF NetworkEdit ?>1<? ELSE ?>0<? ENDIF ?>">
 					<thead>
 					<tr>
+						<td class="ignore-sort">
 						<? IF NetworkEdit ?>
-						<td class="ignore-sort">[<a href="<? VAR URIPrefix TOP ?><? VAR ModPath TOP ?>addnetwork?user=<? VAR Username ESC=URL ?>"><? FORMAT "Add" ?></a>]</td>
```
Comment (DarthGandalf) [inline modules/data/webadmin/tmpl/add_edit_user.tmpl:129]: Why did you remove ignore-sort?

Comment (bashgeek) [reply modules/data/webadmin/tmpl/add_edit_user.tmpl:129]: I restructured that table a bit to make it work properly if it's readonly.
The ignore-sort class is still there, see line 138.
